pandas-stubs/core/groupby/generic.pyi

```diff
@@ -1,21 +1,51 @@
-from typing import overload, List, Iterator
+from typing import overload, Optional, Union, List, Dict, Iterator
+import numpy as _np
 
 from pandas.core.frame import DataFrame
 from pandas.core.series import Series
+from ..frame import _FunctionLike
+
+_str = str  # needed because Series has a property called "str"...
 
 class GroupBy: ...
 
 class DataFrameGroupBy(GroupBy):
     @overload
-    def __getitem__(self, item: str) -> Series: ...
+    def __getitem__(self, item: _str) -> SeriesGroupBy: ...
     @overload
-    def __getitem__(self, item: List[str]) -> DataFrame: ...
+    def __getitem__(self, item: List[_str]) -> DataFrameGroupBy: ...
+    def __getattr__(self, name: _str) -> SeriesGroupBy: ...
     def __iter__(self) -> Iterator: ...
+    def aggregate(
+        self,
+        func: Union[_FunctionLike, List[_FunctionLike], Dict[_str, _FunctionLike]],
+    ) -> DataFrame: ...
+    agg = aggregate
+    def count(self) -> DataFrame: ...
+    def head(self, n: int = ...) -> DataFrame: ...
+    def max(self) -> DataFrame: ...
+    def mean(self) -> DataFrame: ...
+    def median(self) -> DataFrame: ...
+    def min(self) -> DataFrame: ...
+    def nunique(self, dropna: bool = ...) -> DataFrame: ...
+    def quantile(self, q: float = ..., interpolation: str = ...) -> DataFrame: ...
+    def std(self, ddof: int = ...) -> DataFrame: ...
+    def sum(self) -> DataFrame: ...
+    def tail(self, n: int = ...) -> DataFrame: ...
+    def var(self, ddof: int = ...) -> DataFrame: ...
 
 class SeriesGroupBy(GroupBy):
-    def __getitem__(self, item: str) -> Series: ...
+    def __getitem__(self, item: _str) -> Series: ...
+    def count(self) -> Series: ...
+    def head(self, n: int = ...) -> Series: ...
     def max(self) -> Series: ...
     def mean(self) -> Series: ...
     def median(self) -> Series: ...
     def min(self) -> Series: ...
+    def nunique(self, dropna: bool = ...) -> Series: ...
+    def quantile(self, q: float = ..., interpolation: str = ...) -> Series: ...
     def std(self, ddof: int = ...) -> Series: ...
+    def sum(self) -> Series: ...
+    def tail(self, n: int = ...) -> Series: ...
+    def unique(self) -> Series[_np.ndarray]: ...
```

Comment:
This line throws an error in the mypy check. Does it really return a Series of numpy arrays?

Reply:
It does, here's the [documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.SeriesGroupBy.unique.html). And you could try this code:
```
sepal_length = iris.groupby('species').sepal_length
type(sepal_length.unique())
type(sepal_length.unique()[0])
```
I could change that to simple Series return type, should I?

Reply:
I would rather add _np.ndarray to _DType

Reply:
I removed np.ndarray specification for now

Reply:
Ok, I tried your code and is indeed a Series of ndarray. So, yeah, this should be added to `_DType` in `series.pyi`. I'm leaving it up to you if you want to do that. Otherwise I'll merge it as it is right now.

Reply:
I've finally returned the `np.ndarray` specification :)
I also added `list` to the `_DTypes`: it is returned by `Series.str.split()` method, for example.